php/class-edit-post-preview.php

```
@@ -197,4 +206,109 @@ public function make_auto_draft_status_previewable() {
 		global $wp_post_statuses;
 		$wp_post_statuses['auto-draft']->protected = true;
 	}
+
+	/**
+	 * Updates changeset via ajax when preview button is clicked.
+	 *
+	 * @global WP_Customize_Manager $wp_customize
+	 */
+	public function update_post_changeset() {
+		global $wp_customize;
+
+		if ( ! check_ajax_referer( self::UPDATE_CHANGESET_NONCE_ACTION, self::UPDATE_CHANGESET_NONCE, false ) ) {
+			status_header( 400 );
+			wp_send_json_error( 'bad_nonce' );
+		} elseif ( ! current_user_can( 'customize' ) ) {
+			status_header( 403 );
+			wp_send_json_error( 'customize_not_allowed' );
+		} elseif ( ! isset( $_POST['previewed_post'] ) ) {
+			status_header( 400 );
+			wp_send_json_error( 'missing_previewed_post' );
+		} elseif ( empty( $_POST['customize_url'] ) ) {
+			status_header( 400 );
+			wp_send_json_error( 'missing_customize_url' );
+		} elseif ( empty( $_POST['input_data'] ) || ! is_array( $_POST['input_data'] ) ) {
+			status_header( 400 );
+			wp_send_json_error( 'missing_input_data' );
+		}
+
+		$previewed_post_id = intval( wp_unslash( $_POST['previewed_post'] ) );
```

Note that this was previously `absint`. The use of this is function to sanitize integers is being discouraged in core now in favor of `intval` so that negative numbers won't get silently converted into something that could be a real post.